modules/core/src/main/java/org/apache/ignite/internal/processors/metric/GridMetricManager.java

```diff
@@ -300,6 +392,136 @@ public void remove(String regName) {
+            throw new IgniteException(e);
+        }
+        finally {
+            metaLock.writeLock().unlock();
+        }
+    }
+
+    /**
+     * Change {@link HitRateMetric} configuration if it exists.
+     *
+     * @param name Metric name.
+     * @param rateTimeInterval New rate time interval.
+     * @throws IgniteCheckedException If write of configuration failed.
+     * @see HitRateMetric#reset(long, int)
+     */
+    public void configureHitRate(String name, long rateTimeInterval) throws IgniteCheckedException {
+        A.notNullOrEmpty(name, "name");
+        A.ensure(rateTimeInterval > 0, "rateTimeInterval should be positive");
+        A.notNull(metastorage, "Metastorage not ready. Node not started?");
+
+        if (ctx.isStopping())
+            throw new NodeStoppingException("Operation has been cancelled (node is stopping)");
+
+        metastorage.write(metricName(HITRATE_CFG_PREFIX, name), rateTimeInterval);
+    }
+
+    /**
+     * Stores {@link HistogramMetric} configuration in metastorage.
+     *
+     * @param name Metric name.
+     * @param bounds New bounds.
+     * @throws IgniteCheckedException If write of configuration failed.
+     */
+    public void configureHistogram(String name, long[] bounds) throws IgniteCheckedException {
+        A.notNullOrEmpty(name, "name");
+        A.notEmpty(bounds, "bounds");
+        A.notNull(metastorage, "Metastorage not ready. Node not started?");
+
+        if (ctx.isStopping())
+            throw new NodeStoppingException("Operation has been cancelled (node is stopping)");
+
+        metastorage.write(metricName(HISTOGRAM_CFG_PREFIX, name), bounds);
+    }
+
+    /**
+     * Change {@link HitRateMetric} instance configuration.
+     *
+     * @param name Metric name.
+     * @param rateTimeInterval New rateTimeInterval.
+     * @see HistogramMetric#reset(long[])
+     */
+    private void onHitRateConfigChanged(String name, @Nullable Long rateTimeInterval) {
+        if (rateTimeInterval == null)
+            return;
+
+        A.ensure(rateTimeInterval > 0, "rateTimeInterval should be positive");
+
+        HitRateMetric m = find(name, HitRateMetric.class);
+
+        if (m == null)
+            return;
+
+        m.reset(rateTimeInterval);
+    }
+
+    /**
+     * Change {@link HistogramMetric} instance configuration.
+     *
+     * @param name Metric name.
+     * @param bounds New bounds.
+     */
+    private void onHistgoramConfigChanged(String name, @Nullable long[] bounds) {
+        if (bounds == null)
+            return;
+
+        HistogramMetric m = find(name, HistogramMetric.class);
+
+        if (m == null)
+            return;
+
+        m.reset(bounds);
+    }
+
+    /**
+     * @param name Metric name.
+     * @param type Metric type.
+     * @return Metric.
+     */
+    private <T extends Metric> T find(String name, Class<T> type) {
+        A.notNull(name, "name");
+
+        String[] splited = fromFullName(name);
```

Comment:
It's either "splitted" or "split", I would rename it.

Reply:
Renamed to "splitted".